src/main/kotlin/com/hibob/academy/resource/petsResource.kt

```diff
@@ -17,22 +18,26 @@ class PetsResource {
     @Consumes(MediaType.APPLICATION_JSON)
     fun createPet(@RequestBody pets: Pets): Response {
         Response.status(Response.Status.OK).build()
-        return Response.ok("POST OK").build()
+        return Response.ok(pets).build()
     }
 
     @GET
     @Path("/{petId}/type")
     fun getPetType(@PathParam("petId") id: String): Response {
         Response.status(Response.Status.OK).build()
-        return Response.ok("GET OK").build()
+        return Response.ok(listOf(Pets(name = "dog", type = "lab"
+            , company_id = UUID.randomUUID(), date_of_arrival = Date() ))).build()
     }
 
     @PUT
-    @Path("/{petId}/type")
+    @Path("/{petId}")
     @Consumes(MediaType.APPLICATION_JSON)
-    fun updatePetType(@PathParam("petId") id: String, petType: String): Response {
+    fun updatePetType(@PathParam("petId") id: String, @QueryParam("newType") newType: String?): Response {
```

Comment:
not sure if `newType` should be optional here, as if we don't get it - the API has no meaning. I would put it as a mandatory param

Reply:
query param is still optional dont it?
should i change it to path param 
    @Path("/updateType/{petId}/{newType")?

Reply:
you can also mark it as mandatory even if it's a query param, so the framework will throw an exception if the used will not send it. It is also possible to keep in mandatory and throw exception in your code if you don't get it - as without it the API is actually not doing anything